alicloud/resource_alicloud_kvstore_backup_policy.go

```diff
@@ -98,16 +98,12 @@ func resourceAlicloudKVStoreBackupPolicyUpdate(d *schema.ResourceData, meta inte
 	update := false
 	request := r_kvstore.CreateModifyBackupPolicyRequest()
 	request.InstanceId = d.Id()
+	request.PreferredBackupTime = d.Get("backup_time").(string)
+	periodList := expandStringList(d.Get("backup_period").(*schema.Set).List())
+	backupPeriod := fmt.Sprintf("%s", strings.Join(periodList[:], COMMA_SEPARATED))
+	request.PreferredBackupPeriod = backupPeriod
 
-	if d.HasChange("backup_time") {
-		request.PreferredBackupTime = d.Get("backup_time").(string)
-		update = true
-	}
-
-	if d.HasChange("backup_period") {
-		periodList := expandStringList(d.Get("backup_period").(*schema.Set).List())
-		backupPeriod := fmt.Sprintf("%s", strings.Join(periodList[:], COMMA_SEPARATED))
-		request.PreferredBackupPeriod = backupPeriod
+	if d.HasChange("backup_time") || d.HasChange("backup_period") {
 		update = true
 	}
 
```

Comment:
I think `update` can be removed and the update method can be modified as following:
```
func resourceAlicloudKVStoreBackupPolicyUpdate(d *schema.ResourceData, meta interface{}) error {
	if d.HasChange("backup_time") || d.HasChange("backup_period") {
		request := r_kvstore.CreateModifyBackupPolicyRequest()
	        request.InstanceId = d.Id()
	        request.PreferredBackupTime = d.Get("backup_time").(string)
                ....
                if _, err := client.WithRkvClient(func(rkvClient *r_kvstore.Client) (interface{}, error) {
 			return rkvClient.ModifyBackupPolicy(request)
 		}); err != nil {
 			return err
 		}
	}